src/python_routing_v02/troute/routing/diffusive_utils.py

```diff
@@ -387,6 +388,133 @@ def fp_dbcd_map(usgsID2tw=None, usgssDT=None, usgseDT=None, usgspCd=None):
         
     return nts_db_g, dbcd_g
 
+def fp_naturalxsec_map(
+                ordered_reaches,
+                mainstem_headseg_list, 
+                inland_bathyNC, 
+                comid_bathy, 
+                param_df, 
+               #geo_index, 
+                #geo_cols, 
+                #geo_data, 
+                mx_jorder, 
+                mxncomp_g, 
+                nrch_g,
+                dbfksegID):
+    """
+    natural cross section mapping between Python and Fortran using eHydro_ned_cross_sections data
+    
+    Parameters
+    ----------
+    ordered_reaches -- (dict) reaches and reach metadata by junction order
+    mainstem_headseg_list -- (int) a list of link IDs of headsegs of related mainstem reaches 
+    inland_bathyNC --  bathymetry NC data of channel cross section
+    geo_index -- (ndarray of int64s) row indices for geomorphic parameters data array (geo_data)
+    mx_jorder -- (int) max junction order
+    mxncomp_g -- (int) maximum number of nodes in a reach
+    nrch_g -- (int) number of reaches in the network
+    dbfksegID -- (int) segment ID of fake node (=bottom node) of TW reach that hosts downstream boundary 
+                        condition for a network that is being routed. 
+    
+    Returns
+    -------
+    x_bathy_g -- (numpy of float64s) lateral distance of bathy data points
+    z_bathy_g -- (numpy of float64s) elevation of bathy data points
+    size_bathy_g -- (integer) the nubmer of bathy data points of each cross section
+    mxnbathy_g -- (integer) maximum size of bathy data points
+    """    
+    # take only related data from bathy NC file
+    inland_bathyDF = pd.DataFrame()
+    inland_bathyDF['comid'] =  pd.DataFrame(np.ma.filled(inland_bathyNC['comid'][:]))
```

Comment:
I think you can build new data frame columns without the pd.DataFrame of the right hand side. For example....
```
inland_bathyDF['comid'] = np.ma.filled(inland_bathyNC['comid'][:])
```